### ISSUE TITLE
Bug in openephys raw io

### ISSUE THREAD

**Issue by miketrumpis** [NeuralEnsemble/python-neo#1055]
I just tried loading open ephys .continuous (legacy) data through spikeinterface, which calls into neo, and hit this snag. Just on the face, these two lines are probably not what the author had in mind. The `return_index` option returns indices into the argument array `chan_streams_id`, which is the length of all the processor channels. However, those indices are fed into the unique subset array `streams_id`, which is typically length 3. Leads to an `IndexError`

https://github.com/NeuralEnsemble/python-neo/blob/34d4db8fb0dc950dbbc6defd7fb75e99ea877286/neo/rawio/openephysrawio.py#L175-L176

I don't know what the author intended, but perhaps in 475085e62cac908afa1e476cb211ca130538f883 you wanted to add `return_inverse=True` instead?

Fullish traceback

```python
/usr/local/lib/python3.7/dist-packages/spikeinterface/extractors/neoextractors/openephys.py in read_openephys(folder_path, **kwargs)
    106     if np.any([f.endswith('continuous') for f in files]):
    107         #  format = 'legacy'
--> 108         recording = OpenEphysLegacyRecordingExtractor(folder_path, **kwargs)
    109     else:
    110         # format = 'binary'

/usr/local/lib/python3.7/dist-packages/spikeinterface/extractors/neoextractors/openephys.py in __init__(self, folder_path, stream_id)
     36     def __init__(self, folder_path, stream_id=None):
     37         neo_kwargs = {'dirname': folder_path}
---> 38         NeoBaseRecordingExtractor.__init__(self, stream_id=stream_id, **neo_kwargs)
     39 
     40         self._kwargs = dict(folder_path=str(folder_path), stream_id=stream_id)

/usr/local/lib/python3.7/dist-packages/spikeinterface/extractors/neoextractors/neobaseextractor.py in __init__(self, stream_id, **neo_kwargs)
     24     def __init__(self, stream_id=None, **neo_kwargs):
     25 
---> 26         _NeoBaseExtractor.__init__(self, **neo_kwargs)
     27 
     28         # check channel

/usr/local/lib/python3.7/dist-packages/spikeinterface/extractors/neoextractors/neobaseextractor.py in __init__(self, **neo_kwargs)
     14         neoIOclass = eval('neo.rawio.' + self.NeoRawIOClass)
     15         self.neo_reader = neoIOclass(**neo_kwargs)
---> 16         self.neo_reader.parse_header()
     17 
     18         assert self.neo_reader.block_count() == 1, \

/usr/local/lib/python3.7/dist-packages/neo/rawio/baserawio.py in parse_header(self)
    183 
    184         """
--> 185         self._parse_header()
    186         self._check_stream_signal_channel_characteristics()
    187 

/usr/local/lib/python3.7/dist-packages/neo/rawio/openephysrawio.py in _parse_header(self)
    174         # and create streams channels (keep natural order 'CH' first)
    175         stream_ids, order = np.unique(chan_stream_ids, return_index=True)
--> 176         stream_ids = stream_ids[order]
    177         signal_streams = [(f'Signals {stream_id}', f'{stream_id}') for stream_id in stream_ids]
    178         signal_streams = np.array(signal_streams, dtype=_signal_stream_dtype)

IndexError: index 105 is out of bounds for axis 0 with size 3
```

**Comment by samuelgarcia** [comment]
I think I did this but I don't rember why, this looks super weird!!
And there are other bug on this IO I really must have to fix it soon.

**Comment by miketrumpis** [comment]
FYI my suggestion doesn't make sense either ;) That looks like it would just give you back the original order.

However, things seem to be working if I comment out the line 176 and call with the `stream_id='CH'` option.

```python
import spikeinterface.extractors as se
r = se.read_openephys('path/to/dataset', stream_id='CH')
```

**Comment by miketrumpis** [comment]
I made a fork with this (very simple) change -- can I PR?

**Comment by samuelgarcia** [comment]
@JuliaSprenger @miketrumpis : I will be back on this soon!!

**Comment by JuliaSprenger** [comment]
@samuelgarcia any news?

**Comment by JuliaSprenger** [comment]
@samuelgarcia Do you want to fix this yourself or do you accept a PR by @miketrumpis ?

**Comment by samuelgarcia** [comment]
Same as https://github.com/SpikeInterface/spikeinterface/issues/329

**Comment by samuelgarcia** [comment]
Hi sorry, for sooooo long delay.
Could you #1095 ?

**Comment by miketrumpis** [comment]
Yeah, it looks functionally identical to the fork I've been using for a little while :)

Have you had exposure to open ephys version 0.6 yet? There appear to be some new file naming conventions, and interesting new corner cases, e.g. "CH{n}" name collisions! I've had to patch things over by parsing the "Continuous_Data.openephys" xml file, and then shimming that output to get consistent outcomes. I do hope there's an easier way that occurs to you!

**Comment by JuliaSprenger** [comment]
@miketrumpis It might be worth putting a dataset of that version in the test dataset collection. Do you have a small (<10Mb) dataset that could be put on [Gin](https://gin.g-node.org/Neuralensemble/ephy_testing_data)?

**Comment by samuelgarcia** [comment]
@JuliaSprenger : what @miketrumpis mention is independant from this discussion.
I propose merge  PR #1095 and close this issue and open a new one.

**Comment by JuliaSprenger** [comment]
Closed via #1095
